src/spriteloader/grf.cpp

```diff
@@ -69,8 +69,8 @@ static bool WarnCorruptSprite(uint8 file_slot, size_t file_pos, int line)
  */
 bool DecodeSingleSprite(SpriteLoader::Sprite *sprite, uint8 file_slot, size_t file_pos, SpriteType sprite_type, int64 num, byte type, ZoomLevel zoom_lvl, byte colour_fmt, byte container_format)
 {
-	AutoFreePtr<byte> dest_orig(MallocT<byte>(num));
-	byte *dest = dest_orig;
+	std::unique_ptr<byte[]> dest_orig(new byte[num]);
```

Comment:
perhaps a std::vector would make more sense here?

Reply:
std::vector will initialize the memory, which is unnecessary here.

Reply:
sounds like it might be a microoptimisation? is it worth it?
```
std::vector<byte> dest_orig;
dest_orig.reserve(num);
```
(then changing all the usage) should be just as effective

Reply:
>  The pointer is such that range [data(); data() + size()) is always a valid range [...]

And `reserve` does not change the size. It might work on common implementations, but it does not have to.

Reply:
I don't like it, but it's better than what's there already, at least. In future I feel like the function could be reworked to use vector more effectively - reserve would ensure that only a single memory allocation was necessary, then the rest could be solved with a few push_back & size calls, as far as I can tell